frontend/src/components/Dialogs/Profile.tsx

```diff
@@ -90,6 +93,18 @@ const ProfileDialog = ({
     copyToClipboard(`http://${host}/ref/${referralCode}`);
   };
 
+  const handleWeblnInvoiceClicked = (e: any) =>{
+    e.preventDefault();
+    if (earnedRewards) {
+      getWebln()
+        .then((webln) => {
+          webln.makeInvoice(earnedRewards).then((response) => {
+            handleSubmitInvoiceClicked(e, response.paymentRequest);
+          })
+        });
```

Comment:
What about the following suggestion to avoid the nested callbacks?

```ts
const webln = await getWebln();

const response = await webln.makeInvoice(earnedRewards);

handleSubmitInvoiceClicked(e, response.paymentRequest)
```

Reply:
If that's the patterns you apply it's fine, but I'm curious now, id there any difference?